src/Groups/matrices/MatGrp.jl

```diff
@@ -588,10 +588,27 @@ end
 
 """
     general_linear_group(n::Int, q::Int)
-    general_linear_group(n::Int, F::FqNmodFiniteField)
+    general_linear_group(n::Int, F::Ring)
```

Comment:
If `F` is a ring now, perhaps rename it to `R` and also adjust the text in line 594 which specifically talks about `F` being a field?

Same applies elsewhere